File: backend/app/models/library.py
"""RSS Library 관련 Pydantic 모델"""

from pydantic import BaseModel
from typing import Optional, List, Literal
from datetime import datetime


class FeedSource(BaseModel):
    """피드 소스 정보"""
    site_id: Optional[str] = None
    site_name: str
    site_url: str


class LibraryFeedItem(BaseModel):
    """RSS Library 피드 아이템"""
    id: str
    type: Literal['project', 'rss_post']  # 프로젝트 또는 RSS 게시물
    title: str
    title_original: Optional[str] = None  # 원본 제목 (영문 등)
    source: FeedSource
    keywords: List[str] = []
    summary: str
    published_at: Optional[datetime] = None
    url: str
    has_cardnews: bool  # 카드뉴스 생성 여부
    project_id: Optional[str] = None  # 카드뉴스 프로젝트 ID (있는 경우)
    status: Optional[Literal['draft', 'summarized', 'completed']] = None
    is_new: bool = False  # 24시간 이내 여부
    
    class Config:
        from_attributes = True


class LibraryFeedResponse(BaseModel):
    """RSS Library 피드 응답"""
    total: int
    page: int
    page_size: int
    items: List[LibraryFeedItem]


class CreateCardnewsRequest(BaseModel):
    """RSS 게시물에서 카드뉴스 생성 요청"""
    rss_post_id: str
    site_id: str
    url: str
    title: str
    content: str


class CreateCardnewsResponse(BaseModel):
    """카드뉴스 생성 응답"""
    project_id: str
    status: Literal['draft', 'summarized', 'completed']

File: frontend/lib/api/library.ts
/**
 * RSS Library API 클라이언트
 */

import axios from 'axios';

const API_URL = process.env.NEXT_PUBLIC_API_URL || 'http://localhost:8000';

export interface FeedSource {
  site_id?: string | null;
  site_name: string;
  site_url: string;
}

export interface LibraryFeedItem {
  id: string;
  type: 'project' | 'rss_post';
  title: string;
  source: FeedSource;
  keywords: string[];
  summary: string;
  published_at: string | null;
  url: string;
  has_cardnews: boolean;
  project_id: string | null;
  status: 'draft' | 'summarized' | 'completed' | null;
  is_new: boolean;
}

export interface LibraryFeedResponse {
  total: number;
  page: number;
  page_size: number;
  items: LibraryFeedItem[];
}

export interface CreateCardnewsRequest {
  rss_post_id: string;
  site_id: string;
  url: string;
  title: string;
  content: string;
}

export interface CreateCardnewsResponse {
  project_id: string;
  status: 'draft' | 'summarized' | 'completed';
}

/**
 * RSS Library 통합 피드 조회
 */
export async function getLibraryFeed(params: {
  site_id?: string;
  start_date?: string;
  end_date?: string;
  keyword?: string;
  year_month?: string;
  page?: number;
  page_size?: number;
}): Promise<LibraryFeedResponse> {
  const queryParams = new URLSearchParams();
  
  if (params.site_id) queryParams.append('site_id', params.site_id);
  if (params.start_date) queryParams.append('start_date', params.start_date);
  if (params.end_date) queryParams.append('end_date', params.end_date);
  if (params.keyword) queryParams.append('keyword', params.keyword);
  if (params.year_month) queryParams.append('year_month', params.year_month);
  if (params.page) queryParams.append('page', params.page.toString());
  if (params.page_size) queryParams.append('page_size', params.page_size.toString());
  
  const response = await axios.get<LibraryFeedResponse>(
    `${API_URL}/api/library/feed?${queryParams.toString()}`
  );
  
  return response.data;
}

/**
 * RSS 게시물에서 카드뉴스 생성
 */
export async function createCardnewsFromFeed(
  request: CreateCardnewsRequest
): Promise<CreateCardnewsResponse> {
  const response = await axios.post<CreateCardnewsResponse>(
    `${API_URL}/api/library/create-cardnews`,
    request
  );
  
  return response.data;
}

